File: src/services/analytics.py
import os
import requests
import logging

logger = logging.getLogger()

# ─────────────────────────────────────────────
# Cliente para la Analytics API
# (modulo GPT+WhatsApp desarrollado por el reemplazo de Waldo)
#
# NOTA: Esta es una integracion pendiente. Los parametros exactos
# del request/response deben coordinarse cuando el modulo este listo.
# Por ahora se asume la siguiente interfaz como placeholder.
# ─────────────────────────────────────────────

def call_analytics_api(user_id: str, message: str, context: list) -> str:
    """
    Envia el mensaje anonimizado a la Analytics API y retorna la respuesta del GPT.

    Args:
        user_id: ID unico del egresado (nunca datos personales reales)
        message: Texto del usuario ya anonimizado
        context: Historial reciente de la conversacion

    Returns:
        Respuesta de texto generada por GPT
    """
    api_url = os.environ.get("ANALYTICS_API_URL")
    api_key = os.environ.get("ANALYTICS_API_KEY")

    if not api_url:
        logger.error("ANALYTICS_API_URL no configurada")
        return _fallback_response()

    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {api_key}"
    }

    payload = {
        "user_id": user_id,
        "message": message,
        "context": context
        # TODO: ajustar payload segun especificacion del modulo de Waldo
    }

    try:
        response = requests.post(
            f"{api_url}/chat",
            json=payload,
            headers=headers,
            timeout=25
        )
        response.raise_for_status()
        data = response.json()

        # TODO: ajustar key de respuesta segun especificacion del modulo
        return data.get("response", _fallback_response())

    except requests.Timeout:
        logger.error("Timeout al llamar a Analytics API")
        return _fallback_response()
    except requests.RequestException as e:
        logger.error(f"Error llamando a Analytics API: {str(e)}")
        return _fallback_response()


def _fallback_response() -> str:
    """Respuesta de emergencia cuando la API no esta disponible."""
    return (
        "En este momento no puedo procesar tu consulta. "
        "Por favor intenta nuevamente en unos minutos o escribe 'agente' "
        "para ser atendido por una persona."
    )

File: src/services/analytics.js
'use strict';

const axios = require('axios');

// ─────────────────────────────────────────────
// Cliente para la Analytics API
// (modulo GPT+WhatsApp desarrollado por el reemplazo de Waldo)
//
// NOTA: Esta es una integracion pendiente. Los parametros exactos
// del request/response deben coordinarse cuando el modulo este listo.
// Por ahora se asume la siguiente interfaz como placeholder.
// ─────────────────────────────────────────────

const FALLBACK_RESPONSE =
  'En este momento no puedo procesar tu consulta. ' +
  "Por favor intenta nuevamente en unos minutos o escribe 'agente' " +
  'para ser atendido por una persona.';

/**
 * Envia el mensaje anonimizado a la Analytics API y retorna la respuesta del GPT.
 *
 * @param {string} userId   - ID unico del egresado (nunca datos personales reales)
 * @param {string} message  - Texto del usuario ya anonimizado
 * @param {Array}  context  - Historial reciente de la conversacion
 * @returns {Promise<string>} Respuesta de texto generada por GPT
 */
const callAnalyticsApi = async (userId, message, context) => {
  const apiUrl = process.env.ANALYTICS_API_URL;
  const apiKey = process.env.ANALYTICS_API_KEY;

  if (!apiUrl) {
    console.error('ANALYTICS_API_URL no configurada');
    return FALLBACK_RESPONSE;
  }

  const headers = {
    'Content-Type': 'application/json',
    Authorization: `Bearer ${apiKey}`,
  };

  const payload = {
    user_id: userId,
    message,
    context,
    // TODO: ajustar payload segun especificacion del modulo de Waldo
  };

  try {
    const res = await axios.post(`${apiUrl}/chat`, payload, {
      headers,
      timeout: 25000,
    });
    // TODO: ajustar key de respuesta segun especificacion del modulo
    return res.data.response || FALLBACK_RESPONSE;
  } catch (err) {
    if (err.code === 'ECONNABORTED') {
      console.error('Timeout al llamar a Analytics API');
    } else {
      console.error('Error llamando a Analytics API:', err.message);
    }
    return FALLBACK_RESPONSE;
  }
};

module.exports = { callAnalyticsApi };
